### PR TITLE
Add option to skip more frames using report

### PR DESCRIPTION
**Pull request by bz2**
New options parameter on report() for per-call config.

Add skipLocalFrames option to trim the head of a locally generated
stacktrace to remove more than one level.

Enforce valid jsdoc using eslint and add params for test helpers.

Add documentation for passing options to report().

### CODE REVIEW

**Comment by steren** [comment]
Thanks.
What is the use case for skipping frames? Can you add this to the README?

I suppose it is to remove frames of local error reporter frameworks, in which case, would it be better to be able to exclude frames based on some function or filename pattern, instead of an hard coded number?

**Comment by bz2** [comment]
Yes, this is a similar use case to filtering based on frame location. See the README change for an example.

There *is* an options argument for `StackTrace.fromError()` that can be used to override its internal filter, with two niggles:
* The current module-level filtering relies on a naming scheme being used for all functions.
* The default filter is not exported, so would need to be copied to add to the logic.

Basically, this change is a low-friction way of making logged stacks a bit nicer, rather than a complete solution.
